### PR TITLE
Port changes in external-data example app from branch dev/external-data-prototyping into main

### PR DESCRIPTION
**Pull request by ghoshkaj**
## Description

This PR ports over all the changes to the UI and a mock signal from the dev branch (https://github.com/microsoft/FluidFramework/tree/dev/external-data-prototyping) into the main branch to keep the two branches as in sync as possible.

This port completes the loop of making a change to the external data through to the client receiving the signal about it and accepting the changes.

## Breaking Changes

No breaking changes have been ported over, because the dev branch is specifically used for those.

## Reviewer Guidance

This PR contains a lot of changes from https://github.com/microsoft/FluidFramework/tree/dev/external-data-prototyping with the notable difference that the signal from Alfred has been replaced with a submitSignal in the AppModel and that is called in the DebugView "Save Changes" button. There are comments around each of the places where this PR introduces this workaround. 

Also, because we cannot port over the removal filtering out local changes in Collaborative Input (https://github.com/microsoft/FluidFramework/pull/13732/files#diff-052a365945547f8ef89caad6901380a223102046afd565c2c32dfa3a0e52e1a7R60) , currently this PR does not update the text view in the client app after the changes have been accepted and written to SavedData.

mock_signal

### CODE REVIEW

File: examples/hosts/app-integration/external-data/src/model/taskList.ts [hunk 1]
```diff
@@ -156,7 +184,7 @@ export class TaskList extends DataObject implements ITaskList {
         if (this._draftData?.get(id) === undefined) {
             return;
         }
-        const newTask = new Task(id, nameSharedString, prioritySharedCell);
+        const newTask = new Task(id, nameSharedString, prioritySharedCell, undefined, undefined, undefined);
```
Comment (ChumpChief) [inline examples/hosts/app-integration/external-data/src/model/taskList.ts:187]: Should this pull from savedData rather than using undefined?

**Review by ChumpChief — APPROVED** [review]
Approving since I think this is mostly about the port to main - some of the feedback from the prior PR is probably still worth considering though :)

File: examples/hosts/app-integration/external-data/src/model/taskList.ts (hunk 1 above)
Comment (ghoshkaj) [reply examples/hosts/app-integration/external-data/src/model/taskList.ts:187]: Oh good point.

File: examples/hosts/app-integration/external-data/src/model-interface/index.ts [hunk 2]
```diff
@@ -32,21 +32,32 @@ export interface IAppModel extends IEventProvider<IAppModelEvents> {
...
-    (event: "nameChanged" | "priorityChanged", listener: () => void);
+    * Emitted when the name has changed.
+    */
+   (event: "nameChanged", listener: () => void);
+   /**
+    * Emitted when the priority has changed to either the incoming value or to the NONE default value.
+    */
+   (event: "priorityChanged", listener: () => void);
+   /**
+    * Emitted when the incomingName has to either the incoming value or to the NONE default value.
```
Comment (alexvy86) [inline examples/hosts/app-integration/external-data/src/model-interface/index.ts:54]:
```suggestion
    * Emitted when the incomingName has changed to either the incoming value or to the NONE default value.
```

**Review by WayneFerrao — APPROVED** [review]

File: examples/hosts/app-integration/external-data/src/model/taskList.ts [hunk 3]
```diff
@@ -14,36 +14,76 @@ import { customerServicePort } from "../mock-service-interface";
...
+        this._incomingPriority = newValue;
+        this.emit('incomingPriorityChanged');
+    }
+    public constructor(
+        private readonly _id: string,
+        private readonly _name: SharedString,
+        private readonly _priority: ISharedCell<number>,
+        private _incomingName: string | undefined,
+        private _incomingPriority: number | undefined,
+        private _incomingType: string | undefined,
```
Comment (alexvy86) [inline examples/hosts/app-integration/external-data/src/model/taskList.ts:57]: I might just not be getting how the model works, but having the incoming values as constructor parameters feels weird? When creating a task locally they wouldn't be required, and when dealing with incoming changes I'd expect that I'd be applying them to an existing task through its setters.

**Review by alexvy86 — COMMENTED** [review]
Haven't looked at all of it yet, but a few small things.

File: examples/hosts/app-integration/external-data/src/model/taskList.ts [hunk 4]
```diff
@@ -335,13 +402,32 @@ export class TaskList extends DataObject implements ITaskList {
 			}
 		});
 
+		this._draftData.on("valueChanged", (changed) => {
```
Comment (Josmithr) [inline examples/hosts/app-integration/external-data/src/model/taskList.ts:405]: Looks like we're registering this same code twice (see lines above)? Merge conflict issue maybe?

**Pull request merged**